week1/homework/src/server.js

```diff
@@ -5,13 +5,38 @@ const http = require('http');
 /* `createServer` MUST return an instance of `http.Server` otherwise the tests
  * will fail.
  */
+
 function createServer(port) {
   let state = 10;
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
-  });
+    if (request.url.startsWith('/state')) {
+      response.writeHead(200, { 'Content-Type': 'application/json' });
```

Comment:
Hi Ramazan - I am going to be a bit strict starting now as we are getting into advanced topics and it is important that we get the basics right.

Over here - if you look closely, the `{ 'Content-Type': 'application/json' }` is being repeated in all the if-elseif-else scenarios. This means all the response types are the same. Thus, you can move this object out in a constant and then use that variable in the `writeHead`.

Reply:
Thanks to strict @yash-kapila  you can be more strict on it.
I have done it.